### PR TITLE
Initial merge and update of conda-forge recipe.

### PR DESCRIPTION
**Pull request by ScottBailey**
# Changes

Changes:
- Initial import from conda-forge.
- Update version from `14.2.3` to `14.2.4`. (Minor bugfixes.)
- Prefer ninja only (reduce maintenance costs and as a recipe fix).

# Review Information

## Upstream Changes

https://github.com/DaanDeMeyer/reproc/blob/main/CHANGELOG.md

2 memory leak bugfixes and one a non-breaking API interface improvement.

Worth taking, I believe.

## Issues

https://github.com/DaanDeMeyer/reproc/issues

Nothing that should stop this update.

## Pins and Requireds

This is a C and C++ library. The only dependency seems to be the
system's threading library. No specific version is called out.

# Clossing Comments

This is a simple import and update, the most complicated part is the
switch to ninja.

### CODE REVIEW

**Review by tobijk — APPROVED** [review]
Wowser, nice revamp!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
-    sha256: ea87db7a780a81bc27bce3e6c8e5f31b831fb8a1d06c6b5c2e1c2554d1c55e00
+    sha256: {{ sha_hash }}
     patches:
       - apple_gettime.patch  # [osx]
 
 build:
-  number: 0
+  number: {{ build_number }}
+  missing_dso_whitelist:  # [s390x]
+    - $RPATH/ld64.so.1    # [s390x]  Something is slightly amiss in the s390x build process.
```
Comment (tobijk) [inline recipe/meta.yaml:19]: I think this happens because conda-build does not understand that the linker is part of the sysroot.

**Review by tobijk — APPROVED** [review]

File: recipe/build_cpp.sh [hunk 2]
```diff
@@ -1,22 +1,32 @@
...
+cd build-cpp || exit 1
 
 if [[ "$PKG_NAME" == *static ]]; then
     BUILD_TYPE="-DBUILD_SHARED_LIBS=OFF"
 else
     BUILD_TYPE="-DBUILD_SHARED_LIBS=ON"
 fi
 
-cmake ${CMAKE_ARGS} .. \
-      -DCMAKE_BUILD_TYPE=Release \
```
Comment (jaimergp) [inline recipe/build_cpp.sh:11]: I think we need to add this back @ScottBailey @tobijk

Comment (jezdez) [reply recipe/build_cpp.sh:11]: I opened https://github.com/AnacondaRecipes/reproc-feedstock/pull/2 to see if this fixes it.

**Pull request merged**